### PR TITLE
Add support for overriding session save conditions per request

### PR DESCRIPTION
**Pull request by evan-king**
Option for resolving issue https://github.com/koajs/generic-session/issues/88

### CODE REVIEW

File: lib/session.js [hunk 1]
```diff
@@ -214,6 +228,14 @@ module.exports = function (options) {
    *   if session is modified, update cookie and store
    */
   function *refreshSession (session, originalHash, isNew) {
+    if(this._sessionSave === false) {
+      return debug('session save disabled');
+    }
+    if(this._sessionSave === true) {
+      debug('session save forced');
+      return yield saveNow.call(this, this.sessionId, session);
```
Comment (dead-horse) [inline lib/session.js]: if session was set to `null`, should we save it here?

File: lib/session.js [hunk 2]
```diff
@@ -236,15 +258,21 @@ module.exports = function (options) {
...
     debug('session modified');
+    
+    yield saveNow.call(this, this.sessionId, session);
+    
+  }
+  
+  function *saveNow(id, session) {
     compatMaxage(session.cookie);
-
+
```
Comment (dead-horse) [inline lib/session.js]: remove extra spaces

File: lib/session.js [hunk 3]
```diff
@@ -269,6 +297,8 @@ module.exports = function (options) {
     if (!result) {
       return yield next;
     }
+    
+    addCommonAPI.call(this);
```
Comment (dead-horse) [inline lib/session.js]: move `session` getter/setter and `regeneraSession` to addCommonAPI.

File: lib/session.js (hunk 1 above)
Comment (evan-king) [inline lib/session.js]: Hmm...should probably preserve the session destroying behavior.  I'll play with pushing it below the delete block.

**Comment by dead-horse** [comment]
1.11.0

**Pull request merged**